### PR TITLE
Update for new Hanami source dirs

### PR DESCRIPTION
**Pull request by timriley**
Summary of the changes here:

- Removed `lib/app_prototype/entities/` dir — we're not going to be encouraging people to create application-wide base entity classes by default (this is based on my recent conversations with you, @jodosha; am I understanding your desires correctly here?)
- Removed all `#auto_register: false` magic comments from files in `lib/app_prototype/`, since this is no longer a container component dir (i.e. it's not being auto-registered)
- Added a `config/boot/view.rb` file to register `"view.context"` in the application container — this is to provide a place for basic application-wide view behaviour, which I think may still be useful — see my inline comment below
- Moved `actions/`, `entities/` and `views/` source directories into the root of the `slices/main/` slice directory.
- Moved all other main slice source files from `slices/main/lib/main/` to just `slices/main/lib/`
- Removed all the `entities.rb` and `parts.rb` files, whose only job was to open a matching Ruby module, then loop over all the source files in that namespace and require them — both of these jobs are handled by Zeitwerk now!
- Tweak the logger setup to use the rejigged logger settings (this should go away anyway once we figure out the better zero-config defaults, anyway)

### CODE REVIEW

File: lib/app_prototype/action.rb [hunk 1]
```diff
@@ -1,4 +1,3 @@
-# auto_register: false
```
Comment (timriley) [inline lib/app_prototype/action.rb:1]: We no longer need these magic comments inside `lib/app_prototype/` because this is no longer configured as a component dir for the application container.

File: slices/main/lib/repository.rb [hunk 2]
```diff
@@ -1,7 +1,7 @@
+# auto_register: false
 # frozen_string_literal: true
 
 require "app_prototype/repository"
-require_relative "entities"
```
Comment (timriley) [inline slices/main/lib/repository.rb:4]: No longer need lines like this (and the corresponding files) because Zeitwerk will autoload these for us! (This change could have been introduced at the time of alpha2, FWIW).

File: config/boot/view.rb [hunk 3]
```diff
@@ -0,0 +1,7 @@
+# frozen_string_literal: true
+
+Hanami.application.register_bootable :view do |container|
+  start do
+    register "view.context", AppPrototype::View::Context.new
+  end
+end
```
Comment (timriley) [inline config/boot/view.rb:7]: This one is a bit tricky. Now that we're no longer auto-registering from `lib/app_prototype/`, we need to do this if we want that application-wide view context class to be picked up when rendering views from within actions (Hanami actions will look for a `"view.context"` component and use that, if registered, to pass to the views).

I know that we want to keep as much behaviour as possible separated into the slices, but I can still see there being a benefit in allowing users to define some basic behaviour that they want _all_ their application slices to use when rendering views from actions.

I think we can ultimately address this by moving this boot file out from the application template and into the framework itself, just like we do with `lib/hanami/application/container/boot/logger.rb`.

For the time being, however, I think we should make this simpler localised change within the template, so we can buy ourselves some time to think about the best longer-term arrangement for this.

Comment (jodosha) [reply config/boot/view.rb:7]: @timriley

> I know that we want to keep as much behaviour as possible separated into the slices, but I can still see there being a benefit in allowing users to define some basic behaviour that they want all their application slices to use when rendering views from actions.

I agree.

> I think we can ultimately address this by moving this boot file out from the application template and into the framework itself, just like we do with `lib/hanami/application/container/boot/logger.rb`.

Probably we should split the problem in two:

1 Generate the concrete class in the app: `lib/app_prototype/view/context.rb` to define `AppPrototype::View::Context`.
This because we want to allow developers to add **custom behavior** to that context.


2 Register an instance of that class as `view.context`.
This should be done by the framework internals, similar to what we do with logger.
In other words, the registration isn't visible for the final user.
This implies that `config/boot/view.rb` shouldn't be part of the template and generators.

**Review by jodosha — APPROVED** [review]
@timriley Very nice. 👏  Left a comment about view context, let's clarify that before to merge this PR. 😉
